### PR TITLE
pkgs/tools/wayland: enable strictDeps

### PR DESCRIPTION
**Pull request by Artturin**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by twitchyliquid64 — APPROVED** [review]

**Review by Mindavi — APPROVED** [review]
wdomirror patch has been merged already (https://github.com/NixOS/nixpkgs/pull/172466), otherwise LGTM

File: pkgs/tools/wayland/wdomirror/default.nix [hunk 1]
```diff
@@ -20,18 +21,19 @@ stdenv.mkDerivation {
...
-  nativeBuildInputs = [ meson ninja pkg-config wayland-protocols ];
-
-  buildInputs = [ wayland ];
-
   patches = [
     # https://github.com/progandy/wdomirror/pull/7
-    (fetchurl {
+    (fetchpatch {
       url = "https://github.com/progandy/wdomirror/commit/142632208e9ea2b4a4ebd784532efdb8cad7b87c.patch";
       hash = "sha256-z6/8q2vOtmlGFbDVG5BVWWbLQT8kBvJXT9/oZkWS9gw=";
```
Comment (primeos) [inline pkgs/tools/wayland/wdomirror/default.nix:28]: Does this really not change the hash? (IIRC `fetchpatch` does quite some normalization.)

**Review by primeos — COMMENTED** [review]
The diff looks fine to me (apart from that hash?) but I only gave it a quick look. Thanks for those improvements :)

File: pkgs/tools/wayland/wdomirror/default.nix (hunk 1 above)
Comment (primeos) [reply pkgs/tools/wayland/wdomirror/default.nix:28]: Yeah, I do get `sha256-MG71IEwRAjjacAkRoB7Tn45+FbY7LAqTDkVJkoWuQUU=` as new hash.

**Review by primeos — CHANGES_REQUESTED** [review]

File: pkgs/tools/wayland/wdomirror/default.nix (hunk 1 above)
Comment (Artturin) [reply pkgs/tools/wayland/wdomirror/default.nix]: changed it, thanks

**Comment by Synthetica9** [comment]
@ofborg test sway

**Pull request merged**